hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -215,23 +215,25 @@ protected Object getInternal(Data key) {
                 return fromNearCache;
             }
         }
+
+        Data result = null;
         // todo action for read-backup true is not well tested.
         if (mapConfig.isReadBackupData()) {
-            final Object fromBackup = readBackupDataOrNull(key);
-            if (fromBackup != null) {
-                return fromBackup;
-            }
+            result = readBackupDataOrNull(key);
+        }
+
+        if (result == null) {
+            final GetOperation operation = new GetOperation(name, key);
+            operation.setThreadId(ThreadUtil.getThreadId());
+            result = (Data) invokeOperation(key, operation);
         }
-        final GetOperation operation = new GetOperation(name, key);
-        operation.setThreadId(ThreadUtil.getThreadId());
-        final Data value = (Data) invokeOperation(key, operation);
 
         if (nearCacheEnabled) {
             if (notOwnerPartitionForKey(key) || cacheKeyAnyway()) {
-                return putNearCache(key, value);
+                return putNearCache(key, result);
```

Comment:
we do not need to put all `result` here, only put `result` into `near-cache` if the relevant formats are the case as you stated in the PR comment, so if the `IMap` is `BINARY` and `near-cache` is `OBJECT` you can put the value read from backup into `near-cache` otherwise there will be unnecessary memory consumption.